### ISSUE TITLE
TrustSvd  run err  on Epinions datebase,but SocialMF  is ok 

### ISSUE THREAD

**Issue by Gunther17** [guoguibing/librec#253]
`H:\librec-2.0.0\librec-2.0.0\bin>librec rec -exec -conf ../core/src/main/resources/rec/context/rating/trustsvd-test.properties
18/04/25 22:50:33 INFO RecommenderJob: Splitter info: the index of kcv splitter times is 1
18/04/25 22:50:33 INFO TextDataConvertor: Dataset: ../data/Epinions/rating
18/04/25 22:50:33 INFO TextDataConvertor: All dataset files [..\data\Epinions\rating\ratings_data.txt]
18/04/25 22:50:33 INFO TextDataConvertor: All dataset files size 8650119
18/04/25 22:50:33 INFO TextDataConvertor: Now loading dataset file ratings_data
18/04/25 22:50:36 INFO TextDataModel: Transform data to Convertor successfully!
18/04/25 22:50:37 INFO TextDataModel: Split data to train Set and test Set successfully!
18/04/25 22:50:37 INFO TextDataModel: Data size of training is 531857
18/04/25 22:50:37 INFO TextDataModel: Data size of testing is 132965
18/04/25 22:50:37 INFO TextDataModel: Transform data to Feature successfully!
18/04/25 22:50:38 INFO TrustSVDRecommender: Job Setup completed.
18/04/25 22:50:48 INFO TrustSVDRecommender: TrustSVDRecommender iter 1: loss = 765016.601128871, delta_loss = -765016.6
18/04/25 22:50:59 INFO TrustSVDRecommender: TrustSVDRecommender iter 2: loss = 439621.70791923127, delta_loss = 325394.9
18/04/25 22:51:09 INFO TrustSVDRecommender: TrustSVDRecommender iter 3: loss = 411665.3813948281, delta_loss = 27956.326
18/04/25 22:51:19 INFO TrustSVDRecommender: TrustSVDRecommender iter 4: loss = 624375.0526194524, delta_loss = -212709.67
18/04/25 22:51:29 INFO TrustSVDRecommender: TrustSVDRecommender iter 5: loss = 2568761.001471944, delta_loss = -1944386.0
1**8/04/25 22:51:39 INFO TrustSVDRecommender: TrustSVDRecommender iter 6: loss = NaN, delta_loss = NaN
Exception in thread "main" net.librec.common.LibrecException: Loss = NaN or Infinity: current settings does not fit the recommender! Change the settings and try again!
        at net.librec.recommender.AbstractRecommender.isConverged(AbstractRecommender.java:474)
        at net.librec.recommender.context.rating.TrustSVDRecommender.trainModel(TrustSVDRecommender.java:306)
        at net.librec.recommender.AbstractRecommender.recommend(AbstractRecommender.java:221)
        at net.librec.job.RecommenderJob.executeRecommenderJob(RecommenderJob.java:144)
        at net.librec.job.RecommenderJob.runJob(RecommenderJob.java:93)
        at net.librec.tool.driver.RecDriver.run(RecDriver.java:84)
        at net.librec.tool.driver.RecDriver.main(RecDriver.java:111)**

H:\librec-2.0.0\librec-2.0.0\bin>`

**Comment by SunYatong** [comment]
The exception has already told you what to do : "current settings does not fit the recommender. Change the settings and try again."

**Comment by Gunther17** [comment]
thank you@SunYatong  [https://github.com/guoguibing/librec/issues/115](url)
